tox.ini

```
@@ -63,4 +63,14 @@ deps =
 commands =
   black {posargs:src tests tools}
 passenv =
-  {[testenv:common]passenv}
\ No newline at end of file
+  {[testenv:common]passenv}
+[testenv:{,py310-,py311-,py312-}pytest]
+description = Run pytest
+setenv =
+    PYTHONPATH = {toxinidir}/src
+deps =
+  pytest
```

```suggestion
  -rrequirements.txt
```
And then add `pytest` to requirements.txt.
That's a better way, requirements.txt becomes one source of truth.

---

Remember to add pytest you removed from here to requirements.txt file!